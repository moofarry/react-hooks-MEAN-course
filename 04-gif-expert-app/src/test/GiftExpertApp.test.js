import React from "react";
import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe("test in GiftExpertApp.", () => {
  test("should be show the component", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should be  show a list categories", () => {
    const categories = ["Naruto", "Dragon"];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    //expect(wrapper.find("GifExpertApp").length).toBe(categories.length);
  });
});
