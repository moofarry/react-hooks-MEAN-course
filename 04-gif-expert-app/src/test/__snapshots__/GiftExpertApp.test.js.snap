// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in GiftExpertApp. should be  show a list categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp 
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Naruto"
      key="Naruto"
    />
    <GifGrid
      category="Dragon"
      key="Dragon"
    />
  </ol>
</Fragment>
`;

exports[`test in GiftExpertApp. should be show the component 1`] = `
<Fragment>
  <h2>
    GifExpertApp 
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
